src/swagger-specs/destinations.yaml

```diff
@@ -1443,51 +2702,680 @@ components:
   
     SizeSummary:
       type: object
-      description: Model for capturing data size metrics.
-      title: Size Summary
+      description: Model for capturing data size metrics.
+      title: Size Summary
+      properties:
+        inputBytes:
+          type: integer
+          description: Total bytes in input.
+          title: Input Bytes
+        outputBytes:
+          type: integer
+          description: Total bytes in output.
+          title: Output Bytes
+        skippedBytes:
+          type: integer
+          description: Total bytes which were skipped.
+          title: Skipped Bytes
+        failedBytes:
+          type: integer
+          description: Total bytes which failed.
+          title: Failed Bytes
+        extensions:
+          $ref: '#/components/schemas/Extensions'
+    
+    StatusSummary:
+      type: object
+      description: Model for capturing details related to the status of the dataflow run.
+      title: Status Summary
+      properties:
+        status:
+          type: string
+          description: Status representing the current state of the dataflow run.
+          enum:
+            - Success
+            - Processing
+            - Waiting
+            - Failed
+        errors:
+          type: array
+          description: Info about errors.
+          title: Failed Info
+          items:
+            $ref: '#/components/schemas/Info'
+        extensions:
+          $ref: '#/components/schemas/Extensions'
+
+    ConnectionItem:
+      type: object
+      title: Connection Items
+      description: Parameters specific to connections.
+      required:
+        - id
+        - createdAt
+        - updatedAt
+        - createdBy
+        - updatedBy
+        - createdClient
+        - updatedClient
+        - sandboxId
+        - imsOrgId
+        - name
+        - connectionSpec
+        - state
+        - auth
+        - version
+        - etag
+      properties:
```

Comment:
Using `required` attributes makes sense for describing request payloads, but they don't make sense when they're reused in response payloads (since the concept of some fields being "required" in a response and not others doesn't really track).

When reusing object structures between requests and responses, `required` attributes should only be found when they are being referenced for request payloads. A quick workaround is to create a new definition (e.g. `ConnectionItemWithRequired` that `$refs` this one for the object structure, but also adds the `required` attribute. Then it's just a matter of referencing one definition or the other depending on whether it's a request or response.

Reply:
That is a great point and thanks for the workaround suggestion. I'll take this as an improvement in my next PR (I also want to see if there are other places where we can use `$ref`s).